rs/ledger_suite/icp/ledger/src/main.rs

```diff
@@ -812,6 +855,32 @@ fn pre_upgrade() {
         .expect("failed to write instructions consumed to stable memory");
 }
 
+#[cfg(feature = "upgrade-to-memory-manager")]
+#[export_name = "canister_pre_upgrade"]
+fn pre_upgrade() {
+    let start = dfn_core::api::performance_counter(0);
+    setup::START.call_once(|| {
+        printer::hook();
+    });
+
+    let ledger = LEDGER
```

Comment:
Do we want to log this if it happens?

Reply:
This code in the original pre upgrade hasn't changed for over 3 years - we could consider adding some logging, but I wouldn't do it in this PR.